packages/go_router/lib/src/builder.dart

```diff
@@ -56,10 +56,13 @@ class RouteBuilder {
     Key navigatorKey,
     bool routerNeglect,
   ) {
-    List<Page<dynamic>>? pages;
+    if (matches.isEmpty) {
```

Comment:
Consider adding a comment as to when this scenario happens. My guess is this includes async redirect, which isn't obvious.